plugin/virustotal/vtreport.py

```diff
@@ -0,0 +1,404 @@
+
+__author__ = 'gerardofn@virustotal.com'
+
+from datetime import datetime
+import logging
+import vt
+
+SUPPORTED_FILES = ('PEEXE', 'PEDLL', 'MZ', 'MSI', 'COM',
+                   'COFF', 'ELF', 'LINUX', 'MACHO')
+
+
+class VTReport(object):
+
+  def __init__(self, apikey, file_hash, privileges):
+    global SUPPORTED_FILES
+    client = vt.Client(apikey)
+    general = client.get_object('/files/{}', file_hash)
+    file_type = general.get('type_tag')
+    self.private_api = privileges
+    self.compressed_parents = []  # List of dictionaries
+    self.dropped_files = []       # List of dictionaries
+    self.execution_parents = []   # List of dictionaries
+    self.contacted_ips = {}
+    self.contacted_urls = {}
+    self.contacted_domains = {}
+    self.embedded_domains = {}
+    self.embedded_ips = {}
+    self.embedded_urls = {}
+    self.yara_results = []        # List of dictionaries
+    self.comments = {}  
+    self.behaviour = {}
+    self.filenames = []           # List of strings
+    self.sigma_results = {}
+
+    if file_type.upper() in SUPPORTED_FILES:
+      logging.debug('[VT Report] Processing file type %s.', file_type.upper())
+
+      # Public API Key
+      dets = general.get('last_analysis_stats')
+      total = dets['harmless'] + dets['malicious']
+      total += dets['suspicious'] + dets['undetected']
+      self.detections_malicious = str(dets['malicious'])
+      self.detections_total = str(total)
+
+      if file_type in ('PEEXE', 'PEDLL'):
+        cdate = general.get('creation_date')
+        self.creation = datetime.utcfromtimestamp(cdate)
+        self.creation = self.creation.strftime('%Y-%m-%d %H:%M:%S')
+      else:
+        self.creation = None
+
+      fsdate = general.get('first_submission_date')
+      lsdate = general.get('last_submission_date')
+      ladate = general.get('last_analysis_date')
+      self.fs = datetime.utcfromtimestamp(fsdate).strftime('%Y-%m-%d %H:%M:%S')
+      self.ls = datetime.utcfromtimestamp(lsdate).strftime('%Y-%m-%d %H:%M:%S')
+      self.la = datetime.utcfromtimestamp(ladate).strftime('%Y-%m-%d %H:%M:%S')
+      self.md5 = general.get('md5')
+      self.sha256 = general.get('sha256')
+      self.filenames = general.get('names')
+      self.pe_info = general.get('pe_info')
+      self.packers = general.get('packers')
+      self.tags = general.get('tags')
+      self.yara_results = general.get('crowdsourced_yara_results')
+      self.sigma_results = general.get('sigma_analysis_summary')
+      self.trid = general.get('trid')
+
+      for comment in client.iterator('/files/{}/comments', file_hash, limit=10):
+        self.comments[comment.get('date')] = comment.get('text')
+
+      for behaviour in client.iterator('/files/{}/behaviours',
+                                       file_hash,
+                                       limit=10):
+        self.behaviour[behaviour.get('sandbox_name')] = behaviour
+
+      for dfile in client.iterator('/files/{}/dropped_files',
```

Comment:
This function is really really long, what about splitting it into one function per block / per kind of data, and then just calling these 5 or 6 functions from here?

Say, one for dropped_files, one for behaviour, one for comments, one for comments, etc... make these self._XXX so that it is clear they are private.